packages/zoe/src/contracts/sellItems.js

```diff
@@ -0,0 +1,92 @@
+/* eslint-disable no-use-before-define */
+// @ts-check
+
+import harden from '@agoric/harden';
+import { makeZoeHelpers, defaultAcceptanceMsg } from '../contractSupport';
+
+/** @typedef {import('../zoe').ContractFacet} ContractFacet */
+
+// zcf is the Zoe Contract Facet, i.e. the contract-facing API of Zoe
+export const makeContract = harden(
+  /** @param {ContractFacet} zcf */ zcf => {
+    const { assertKeywords, rejectOffer, checkHook } = makeZoeHelpers(zcf);
+    assertKeywords(harden(['Items', 'Money']));
+
+    const { terms } = zcf.getInstanceRecord();
+    const { pricePerItem } = terms;
+
+    let sellerOfferHandle;
+
+    const sellerOfferHook = offerHandle => {
+      sellerOfferHandle = offerHandle;
+      return defaultAcceptanceMsg;
+    };
+
+    const buyerOfferHook = buyerOfferHandle => {
+      const sellerAllocation = zcf.getCurrentAllocation(sellerOfferHandle);
+      const buyerAllocation = zcf.getCurrentAllocation(buyerOfferHandle);
+      const currentItemsForSale = sellerAllocation.Items;
+      const providedMoney = buyerAllocation.Money;
+
+      const { proposal } = zcf.getOffer(buyerOfferHandle);
+      const wantedItems = proposal.want.Items;
+      const numItemsWanted = wantedItems.extent.length;
+      const amountMaths = zcf.getAmountMaths(['Items', 'Money']);
+      const totalCostExtent = pricePerItem.extent * numItemsWanted;
```

Comment:
I'm not sure what you mean. There's no multiplication in AmountMath

Reply:
Understood, but multiplication is just repeated addition. I'm willing to listen to why a for loop over amountMath addition (which would allow the Price not to be `Nat`) would be worse than presuming `Nat`.

Reply:
Oh, I see. That's a good question. I think since this contract is specifically invoking selling something for money, it makes sense to enforce the fact the "money" digital asset is fungible.  It'd be nice to show the variant as another example though

Reply:
I added an assertion that the moneyIssuer has natMathHelpers.